File: ios/Schoolerz/Data/Mock/MockCommentsRepository.swift
import Foundation

final class MockCommentsRepository: CommentsRepository {
    private var comments: [String: [Comment]] = [:]

    func fetchComments(for postId: String) async throws -> [Comment] {
        try await Task.sleep(for: .milliseconds(300))
        return comments[postId]?.sorted { $0.createdAt < $1.createdAt } ?? []
    }

    func addComment(_ comment: Comment) async throws {
        try await Task.sleep(for: .milliseconds(200))
        comments[comment.postId, default: []].append(comment)
    }
}

File: android/app/src/main/java/com/schoolerz/data/mock/MockCommentsRepository.kt
package com.schoolerz.data.mock

import com.schoolerz.domain.model.Comment
import com.schoolerz.domain.repository.CommentsRepository
import kotlinx.coroutines.delay

class MockCommentsRepository : CommentsRepository {
    private val comments = mutableMapOf<String, MutableList<Comment>>()

    override suspend fun fetchComments(postId: String): Result<List<Comment>> {
        delay(300)
        return Result.success(comments[postId]?.sortedBy { it.createdAt } ?: emptyList())
    }

    override suspend fun addComment(comment: Comment): Result<Unit> {
        delay(200)
        comments.getOrPut(comment.postId) { mutableListOf() }.add(comment)
        return Result.success(Unit)
    }
}
